src/infra/y-redis/helper.ts

```diff
@@ -0,0 +1,57 @@
+import { array, map } from 'lib0';
+import { StreamsMessagesReply } from '../redis/interfaces/stream-message-reply.js';
+import { YRedisMessage } from './interfaces/stream-message.js';
+
+/* This file contains the implementation of the functions,
+    which was copied from the y-redis repository.
+	Adopting this code allows us to integrate proven and
+	optimized logic into the current project.
+	The original code from the `y-redis` repository is licensed under the AGPL-3.0 license.
+	By adhering to the license terms, we ensure that the use of the code from the `y-redis` repository is legally compliant.
+*/
+export const isSmallerRedisId = (a: string, b: string): boolean => {
+	const [a1, a2 = '0'] = a.split('-');
+	const [b1, b2 = '0'] = b.split('-');
+	const a1n = parseInt(a1);
+	const b1n = parseInt(b1);
+
+	return a1n < b1n || (a1n === b1n && parseInt(a2) < parseInt(b2));
+};
+
+export const computeRedisRoomStreamName = (room: string, docid: string, prefix: string): string =>
+	`${prefix}:room:${encodeURIComponent(room)}:${encodeURIComponent(docid)}`;
+
+export const decodeRedisRoomStreamName = (rediskey: string, expectedPrefix: string) => {
+	const match = rediskey.match(/^(.*):room:(.*):(.*)$/);
+	if (match == null || match[1] !== expectedPrefix) {
+		throw new Error(
+			`Malformed stream name! prefix="${match?.[1]}" expectedPrefix="${expectedPrefix}", rediskey="${rediskey}"`,
+		);
+	}
+
+	return { room: decodeURIComponent(match[2]), docid: decodeURIComponent(match[3]) };
+};
+
+export const extractMessagesFromStreamReply = (
+	streamReply: StreamsMessagesReply,
+	prefix: string,
+): Map<string, Map<string, YRedisMessage>> => {
+	const messages = new Map<string, Map<string, YRedisMessage>>();
+
+	streamReply?.forEach((docStreamReply) => {
+		const { room, docid } = decodeRedisRoomStreamName(docStreamReply.name.toString(), prefix);
+		const docMessages = map.setIfUndefined(map.setIfUndefined(messages, room, map.create), docid, () => ({
+			// @ts-ignore
+			lastId: array.last(docStreamReply.messages).id,
+			messages: [] as Uint8Array[],
+		}));
+		docStreamReply.messages?.forEach((m) => {
+			if (m.message.m != null) {
+				// @ts-ignore
```

Comment:
// @ts-ignore ist schon immer relative ultimativ. Evtl. reichen auch einige Ausnahme. Gleiche für Zeile 44.

Reply:
```typscript
export const extractMessagesFromStreamReply = (
	streamReply: StreamMessagesReply,
	prefix: string,
): Map<string, Map<string, YRedisMessage>> => {
	const messages = new Map<string, Map<string, YRedisMessage>>();

	streamReply?.forEach((docStreamReply) => {
		const { room, docid } = decodeRedisRoomStreamName(docStreamReply.name.toString(), prefix);
		const docMessages = map.setIfUndefined(map.setIfUndefined(messages, room, map.create), docid, () => ({
			// @ts-ignore
			lastId: array.last(docStreamReply.messages).id,
			messages: [] as Uint8Array[],
		}));
		docStreamReply.messages?.forEach((m) => {
			if (m.message.m != null) {
				// @ts-ignore
				docMessages.messages.push(m.message.m);
			}
		});
	});

	return messages;
};
```

Wenn ich die Kommentare entferne, bekomme ich :
- in Zeile 44 _(ehemals 45)_ den Fehler :
```
Argument of type 'StreamMessageReply[] | null' is not assignable to parameter of type 'ArrayLike<StreamMessageReply>'.
Type 'null' is not assignable to type 'ArrayLike<StreamMessageReply>'.ts(2345)
 ```
- und in Zeile 49 _(ehemals 51)_ :
```
Argument of type 'RedisKey' is not assignable to parameter of type 'Uint8Array'.
Type 'string' is not assignable to type 'Uint8Array'.ts(2345)
```

Reply:
i pushed a change please review: https://github.com/hpi-schul-cloud/tldraw-server/pull/25/commits/38125e79edb862713a067267e658a7dabb7774c0